### PR TITLE
core: Correctly close sockets

### PR DESCRIPTION
**Pull request by fibonacci-matrix**
Closes: #2353

### CODE REVIEW

File: src/mavsdk/core/tcp_client_connection.cpp [hunk 1]
```diff
@@ -212,7 +199,7 @@ void TcpClientConnection::receive()
         }
 
         if (recv_len < 0) {
-            // This happens on desctruction when close(_socket_fd) is called,
+            // This happens on desctruction when close(_socket_fd.get()) is called,
```
Comment (JonasVautherin) [inline src/mavsdk/core/tcp_client_connection.cpp:202]:
```suggestion
            // This happens on destruction when close(_socket_fd.get()) is called,
```

**Review by julianoes — APPROVED** [review]
Love it, thank you! Keep them coming :smile:

**Comment by julianoes** [comment]
Windows CI isn't happy somehow: https://github.com/mavlink/MAVSDK/actions/runs/10061001774/job/27817265595?pr=2357#step:7:55

File: src/mavsdk/core/socket_holder.h [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    ~SocketHolder() noexcept;
+
+    void reset(int fd) noexcept;
+    void close() noexcept;
+
+    bool empty() const noexcept;
+    int get() const noexcept;
+
+#if defined(WINDOWS)
+    static constexpr int invalid_socket_fd = INVALID_SOCKET;
```
Comment (julianoes) [inline src/mavsdk/core/socket_holder.h:26]: Presumably to fix the Windows warning? :shrug: 

```suggestion
    static constexpr int invalid_socket_fd = static_cast<int>(INVALID_SOCKET);
```

**Comment by fibonacci-matrix** [comment]
It turns out that on 64-bit Windows the socket handle is also 64-bit, unlike POSIX systems where it is always `int`. I will work on it. 

For more details see https://stackoverflow.com/questions/1953639/is-it-safe-to-cast-socket-to-int-under-win64

**Comment by fibonacci-matrix** [comment]
@julianoes , @JonasVautherin I don't have much experience with Windows, could you please check if these lines are necessary in the newly created `.cpp` file?
```
#ifdef WINDOWS
#ifndef MINGW
#pragma comment(lib, "Ws2_32.lib") // Without this, Ws2_32.lib is not included in static library.
#endif
#else
```

**Comment by julianoes** [comment]
@fibonacci-matrix I don't know either. If someone that knows Windows complains, we can fix it.

**Comment by julianoes** [comment]
Should be good to be merged, right @fibonacci-matrix?

**Comment by fibonacci-matrix** [comment]
> Should be good to be merged, right @fibonacci-matrix?

Yes of course, @julianoes

**Pull request merged**